### PR TITLE
docs(module federation): improve module federation docs

### PR DESCRIPTION
**Pull request by burhanuday**
Added a few useful pointers to the module federation plugin docs and also fixed a few grammatical errors

### CODE REVIEW

File: src/content/concepts/module-federation.mdx [hunk 1]
```diff
@@ -22,7 +23,7 @@ related:
 
 ## Motivation
 
-Multiple separate builds should form a single application. These separate builds should not have dependencies between each other, so they can be developed and deployed individually.
+Multiple separate builds should form a single application. These separate builds act like containers and can expose and consume code between builds, creating a single, unified application.
```
Comment (burhanuday) [inline src/content/concepts/module-federation.mdx:26]: `should not have dependencies between each other, so they can be developed and deployed individually` Is a little confusing and makes me thing that dependencies cannot be shared. Not sure what this was originally intended to mean

File: src/content/plugins/module-federation-plugin.mdx [hunk 2]
```diff
@@ -135,7 +136,7 @@ The package name that is used to determine required version from description fil
 
 `false | string`
 
-The required version of the package.
+The required version of the package. A semantic version can also be provided
```
Comment (burhanuday) [inline src/content/plugins/module-federation-plugin.mdx:139]: Meant to say "semver". Could be phrased better. Any ideas?

File: src/content/plugins/module-federation-plugin.mdx [hunk 3]
```diff
@@ -155,18 +156,22 @@ The name of the shared scope.
...
-This hint allows webpack to reject the shared module if version is not valid (defaults to `true` when local fallback module is available and shared module is not a singleton, otherwise `false`, it has no effect if there is no required version specified).
+This hint allows webpack to reject the shared module if version is not valid (defaults to `true` when local fallback module is available and shared module is not a singleton, otherwise `false`, it has no effect if there is no required version specified). Throws a runtime error if the required version is not found.
 
 #### **`version`**
 
 `false | string`
 
 The version of the provided module. It allows webpack to replace lower matching versions, but not higher.
 
+By default, webpack uses the version from the `package.json` file of the dependency
```
Comment (snitin315) [inline src/content/plugins/module-federation-plugin.mdx:173]:
```suggestion
By default, webpack uses the version from the `package.json` file of the dependency.
```

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**